### PR TITLE
refactor(profiling): simplify pprof serialization with event types

### PR DESCRIPTION
**Pull request by jd**
This simplify the logic of setting the data types in the pprof files. This is
not used to discover what's available or not anyway, this won't change the user
experience but simplify the serialization code and testing.

### CODE REVIEW

File: ddtrace/profiling/exporter/pprof.py [hunk 1]
```diff
@@ -361,35 +361,15 @@ def export(self, events, start_time_ns, end_time_ns):
...
-            ("wall-time", "nanoseconds"),
-            ("uncaught-exceptions", "count"),
-            ("lock-acquire", "count"),
-            ("lock-acquire-wait", "nanoseconds"),
-            ("lock-release", "count"),
-            ("lock-release-hold", "nanoseconds"),
-        )
-
-        # Handle StackExceptionSampleEvent
-        if stack.FEATURES["stack-exceptions"]:
```
Comment (Kyle-Verhoog) [inline ddtrace/profiling/exporter/pprof.py:376]: Is this feature flag no longer used? This seems to be the only occurrence. If so, there are a few references that can be cleaned up:

```
tests/profiling/test_main.py
45:        if stack.FEATURES["stack-exceptions"]:

tests/profiling/collector/test_stack.py
204:@pytest.mark.skipif(not stack.FEATURES["stack-exceptions"], reason="Stack exceptions not supported")
229:@pytest.mark.skipif(not stack.FEATURES["stack-exceptions"], reason="Stack exceptions not supported")

tests/profiling/exporter/test_pprof.py
419:        if stack.FEATURES["stack-exceptions"]:
420:            filename = "test-pprof-exporter_tracemalloc+stack-exceptions.txt"
479:    if stack.FEATURES["stack-exceptions"]:

ddtrace/profiling/exporter/pprof.py
376:        if stack.FEATURES["stack-exceptions"]:

ddtrace/profiling/collector/stack.pyx
42:    "stack-exceptions": False,
178:    FEATURES['stack-exceptions'] = True
```

Comment (jd) [reply ddtrace/profiling/exporter/pprof.py:376]: It is used by the tests, like your grep shows. ;)

Comment (Kyle-Verhoog) [reply ddtrace/profiling/exporter/pprof.py:376]: Right but there no longer is any usage in the core code. It should be moved to the tests then?

Comment (jd) [reply ddtrace/profiling/exporter/pprof.py:376]: No, it's just gonna be harder to move to the tests as it uses Cython macro, and it's likely it'll be re-used later to export profiler capabilities to the backend.

**Review by Kyle-Verhoog — APPROVED** [review]

**Pull request merged**